Sources/MuxUploadSDK/Upload/ChunkedFileUploader.swift

```diff
@@ -131,7 +133,43 @@ class ChunkedFileUploader {
                 } else {
                     MuxUploadSDK.logger?.debug("Task finished due to error in state \(String(describing: self.currentState))")
                     let uploadError = InternalUploaderError(reason: error, lastByte: lastReadCount)
+
+                    let lastUpdate: Update?
+                    if case InternalUploadState.uploading(let update) = currentState {
+                        lastUpdate = update
+                    } else {
+                        lastUpdate = nil
+                    }
+
+                    // This modifies currentState, so capture
+                    // the last update first
                     notifyStateFromWorker(.failure(uploadError))
+
+                    // FIXME: Will only work if currentState
```

Comment:
Workaround done in order to minimize changing `ChunkedFileUploader` internals, suggestions for improvement very welcome